src/Utilities.ts

```diff
@@ -105,4 +105,42 @@ export class Utilities<TEightBittr extends FullScreenPokemon> extends GameStartr
 
         return true;
     }
+
+    /**
+     * Function to check if a pokemon should evolve at current level.
+     * 
+     * @param actor   Pokemon to be checked for evolution.
+     * @returns Whether the pokemon is able to evolve at current level.
+     */
+    public shouldEvolve(actor: IPokemon): boolean {
+        let t: string = actor.title.join();
+        let p: any = this.EightBitter.MathDecider.getConstant("pokemon")[t];
+        return actor.level === parseInt(p.evolvesVia.split(" ")[1], 10);
+    }
+
+    /**
+     * Function to evolve a pokemon and subsequently modify their stats
+     * 
+     * @param actor   Pokemon to be evolved
+     * @returns   void
+     */
+    public evolvePokemon(actor: IPokemon): void {
+        let t: string = actor.title.join();
+        let p: any = this.EightBitter.MathDecider.getConstant("pokemon")[t];
+        let evolution: any = p.evolvesInto;
+        actor.title = evolution.toUppercase().split("");
+
+        let evolInfo: any = this.EightBitter.MathDecider.getConstant("pokemon")[evolution.toUppercase()];
+
+        actor.Attack = this.EightBitter.MathDecider.compute("pokemonStatistic", actor, evolInfo.Attack);
```

Comment:
https://github.com/FullScreenShenanigans/FullScreenPokemon/blob/master/src/Settings/Math.ts#L538

You should be able to loop over those instead of hardcoding it, yes?